book/src/pil/types.md

```diff
@@ -9,13 +9,75 @@ The powdr-pil language has the following types:
 - tuple
 - array
 - function type
-- expression
-- constraint
+- `expr` (expression)
+- `constr` (constraint)
+- `!` ("bottom" or "unreachable" type)
+
+> The `col` type is special in that it is not a proper type, but can be used for declaring symbols.
```

Comment:
I'm a bit confused: what's a proper type? Isn't `col` just a type with special rules for how values can be declared (from `int->fe` and `int->fe`)?

Reply:
I can extend it. What I meant was that `col` can only be used at that location and no expression will ever have the type `col` (because it is not a proper type).

Reply:
I mentioned this in another comment, maybe it's worth a chat discussion: it's almost like the `col` and `wit` business is a lower level language inside pil that's arithmetization specific.